tests/test_geography.py

```diff
@@ -11,13 +11,58 @@ def test_point() -> None:
     assert repr(point).startswith("POINT (5.2 40.")
 
 
-def test_linestring() -> None:
-    line = spherely.LineString([(50, 5), (51, 6)])
+@pytest.mark.parametrize(
+    "coords",
+    [
+        [(50, 5), (51, 6)],
+        [spherely.Point(50, 5), spherely.Point(51, 6)],
+    ],
+)
+def test_linestring(coords) -> None:
+    line = spherely.LineString(coords)
     assert line.dimensions == 1
     assert line.nshape == 1
     assert repr(line).startswith("LINESTRING (5 50")
 
 
+@pytest.mark.parametrize(
+    "coords",
+    [
+        [(0, 0), (0, 2), (2, 2), (2, 0)],
+        [spherely.Point(0, 0), spherely.Point(0, 2), spherely.Point(2, 2), spherely.Point(2, 0)],
+    ],
+)
+def test_polygon(coords) -> None:
+    poly = spherely.Polygon(coords)
+    assert poly.dimensions == 2
+    assert poly.nshape == 1
+    assert repr(poly).startswith("POLYGON ((0 0")
+
+
+def test_polygon_error() -> None:
+    with pytest.raises(ValueError, match="loop is not valid.*duplicate vertex.*"):
+        # polygon vertices should be open (duplicate vertex error)
+        spherely.Polygon([(0, 0), (0, 2), (2, 0), (0, 0)])
```

Comment:
Is that something we can control? Because that's different as in shapely, and eg also the WKT shows it as a closed ring (from my other branch using convex_hull, the object is just still a base class Geography object, but the repr already works):
```
In [16]: line = spherely.LineString([(0, 0), (2, 0), (2, 2)])

In [17]: res = spherely.convex_hull(line)

In [18]: res
Out[18]: POLYGON ((0 0, 2 2, 0 2, 0 0))
```

So I would expect that I can recreate such a Polygon from the coordinates shown in the repr

Reply:
Sounds good (was also thinking about that manual check on our side if this is not something that can be controlled otherwise)

Reply:
And this can certainly be left for a follow-up I think, it will be good to have this basic polygon class for other PRs and writing proper tests